qiskit/transpiler/target.py

```diff
@@ -1008,6 +1015,202 @@ def __str__(self):
                 output.write("".join(prop_str_pieces))
         return output.getvalue()
 
+    @classmethod
+    def from_configuration(
```

Comment:
This constructor is more general and takes slightly different information. `convert_to_target` is explicitly to go from a `BackendV1`'s model objects (`BackendConfiguration`, `BackendProperties`, and `PulseDefaults`) while this is to go from the loose objects which are basically the input to `transpile()` and build a `Target` from those. There is definitely overlap as at the end of the day both the `transpile()` arguments and `BackendV1` are expressing the same data (with roughly the same limitations) just in different formats. I think a good follow up to this might be to update `convert_to_target` to leverage this constructor instead of doing it manually. But I feel that's outside the scope of this PR (there is also potential overhead with doing that because we basically have to build a bunch of intermediate objects from the `BackendConfiguration` to match the input here).

Reply:
This method seems to me like encouraging end-user to build `Target` from these legacy data structure. Alternatively you can introduce private factory function in the transpile, so that end-user still cannot build `Target` with these transpiler args, which will be deprecated in future.